torch/jit/_tuples.py

```diff
@@ -0,0 +1,2 @@
+# This is a dummy module where we put all of the known NamedTuple types so that they can be
```

Comment:
btw, is it only for tuples, wouldn't the same problem affect custom classes too? In this case should we call it _types?

Reply:
Classes don't work when deserialized:

```
# foo1.py
import torch

@torch.jit.script
class Foo:
    def __init__(self):
        pass

@torch.jit.script
def foo():
    return Foo()

torch.jit.save(foo, 'f.zip')

```

```
# foo2.py
import torch

loaded = torch.jit.load('f.zip')

print(type(loaded()))

```

```
$ python foo1.py
$ python foo2.py
Traceback (most recent call last):
  File "foo2.py", line 5, in <module>
    print(type(loaded()))
  File "/Users/jamesreed/pytorch/torch/nn/modules/module.py", line 539, in __call__
    result = self.forward(*input, **kwargs)
RuntimeError: classType INTERNAL ASSERT FAILED at ../torch/csrc/jit/pybind_utils.h:579, please report a bug to PyTorch.  (toPyObject at ../torch/csrc/jit/pybind_utils.h:579)
frame #0: c10::Error::Error(c10::SourceLocation, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) + 64 (0x1150e49d0 in libc10.dylib)
frame #1: torch::jit::toPyObject(c10::IValue&&) + 5677 (0x10f0d277d in libtorch_python.dylib)
frame #2: torch::jit::invokeScriptFunctionFromPython(torch::jit::Function&, torch::jit::tuple_slice, pybind11::kwargs, c10::optional<c10::IValue>) + 320 (0x10f16f500 in libtorch_python.dylib)
frame #3: torch::jit::invokeScriptMethodFromPython(torch::jit::script::Method&, torch::jit::tuple_slice, pybind11::kwargs) + 140 (0x10f17138c in libtorch_python.dylib)
frame #4: void pybind11::cpp_function::initialize<torch::jit::script::initJitScriptBindings(_object*)::$_27, pybind11::object, pybind11::args, pybind11::kwargs, pybind11::name, pybind11::is_method, pybind11::sibling>(torch::jit::script::initJitScriptBindings(_object*)::$_27&&, pybind11::object (*)(pybind11::args, pybind11::kwargs), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&)::'lambda'(pybind11::detail::function_call&)::__invoke(pybind11::detail::function_call&) + 227 (0x10f1710d3 in libtorch_python.dylib)
frame #5: pybind11::cpp_function::dispatcher(_object*, _object*, _object*) + 2865 (0x10eda6d21 in libtorch_python.dylib)
<omitting python frames>
frame #24: start + 1 (0x7fff6a1763d5 in libdyld.dylib)
```

Reply:
yeah, but don't we want to make them work in the future? my question is just what the namespace should be really.